encord/storage.py

```diff
@@ -1108,6 +1238,20 @@ def _get_content_type(self, file_path: Union[Path, str], item_type: StorageItemT
             return "application/nifti"
         elif item_type == StorageItemType.DICOM_FILE:
             return "application/dicom"
+        elif item_type == StorageItemType.PLAIN_TEXT:
+            text_mime = mimetypes.guess_type(str(file_path))[0]
+            if text_mime and (
+                text_mime.startswith("text/")
+                or text_mime.startswith("application/json")
+                or text_mime.startswith("application/xml")
+            ):
```

Comment:
Not one for this PR, but in general:

Given the increasing number of modalities (and hence mime types), Would it be worth creating a `MIME_TYPE` String enum? With all the different mime types we sort of expect?
- This would reduce the possibility of silly typo's
- Serves as nice documentation of the mimetypes we support.

Although I also appreciate there might be sooooo many mimetypes, that it might become unwieldy to maintain this enum.

Reply:
Thinking of something like this:

```
class EncordMimeType(StringEnum):
    VIDEO_MP4 = auto()
    AUDIO_MP3 = auto()
    TEXT_JSON = auto()
    TEXT_HTML = auto()
    APPLICATION_JSON = auto()

    @classmethod
    def is_plain_text(cls, mime_type: str) -> bool:
        return (mime_type.startswith(cls.TEXT_HTML)
                or mime_type.startswith(cls.TEXT_JSON))

```

Reply:
Potentially. We don't necessarily need one client side here given that this code is relatively static.
Also your plain text method doesn't handle the newly introduced cases of supporting all textual files.

Reply:
We need to think about how we handle this (current way it not great), but in actual reality there are way too many text MIMEs for that (e.g. basically every programming language is `text/x-{langname}`)